components/earnings/commons/earnings.lua

```diff
@@ -0,0 +1,132 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Earnings
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Earnings = {}
+local MathUtils = require('Module:Math')
+local String = require('Module:StringUtils')
+local Class = require('Module:Class')
+
+---
+-- Entry point for players and individuals
+-- @player - the player/individual for whom the earnings shall be calculated
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+function Earnings.calculateForPlayer(args)
+	local player = args.player
+
+    if String.isEmpty(player) then
+        return 0
+    end
+	player = mw.ext.TeamLiquidIntegration.resolve_redirect(player)
+
+    local money = 0
+    local condition = '([[players_p1::' .. player .. ']] OR'
+        .. '[[players_p2::' .. player .. ']] OR'
+        .. '[[players_p3::' .. player .. ']] OR'
+        .. '[[players_p4::' .. player .. ']] OR'
+        .. '[[players_p5::' .. player .. ']] OR'
+        .. '[[players_p6::' .. player .. ']] OR'
+        .. '[[players_p7::' .. player .. ']] OR'
+        .. '[[players_p8::' .. player .. ']] OR'
+        .. '[[players_p9::' .. player .. ']] OR'
+        .. '[[players_p10::' .. player .. ']] OR'
+        .. '[[participant::' .. player .. ']])'
+
+    money = money + Earnings._calculateIndividualEarnings(condition, args.year, args.mode)
+
+    return MathUtils._round(money)
+end
+
+function Earnings.calculateForTeam(args)
+    local team = args.team
+    if String.isEmpty(team) then
+        return 0
+    end
+	team = mw.ext.TeamLiquidIntegration.resolve_redirect(team)
+
+    local conditions = '([[participant::' .. team .. ']] OR [[extradata_participantteam::' .. team .. ']])'
+
+    if String.isNotEmpty(args.year) then
+        conditions = conditions .. ' AND [[date::>' .. args.year .. '-01-01]] AND [[date::<' .. args.year .. '-12-31]]'
+    end
```

Comment:
Just a though, maybe calculate it for all years once, and then cache the results in a per year basis. This as we're likely to query the results for every single year, and the extradata_team and players_p filtering can be very expensive on the database.

Reply:
Copy paste from the `TO-DO (Follow UP PR)` section above
> Add `Module:WikiInfo` (or some other name) that holds the base wiki infos like startyear of the game of the wiki
> with that we can also move the lpdb.extradata[earningsinXXXX] to commons using this earnings module

yeah my plan is to enable that but we have to have that `Module:WikiInfo` (or whatever name we choose) first to get the start year from which we query onwards